src/common/mfu_daos.c

```diff
@@ -2556,38 +2575,160 @@ static int cont_serialize_props(struct hdf5_args *hdf5,
                                 daos_handle_t cont)
 {
     int                     rc = 0;
-    daos_prop_t             *prop_query;
-    uint32_t                i;
-    char                    cont_str[64];
-    int                     len = 0;
+    daos_prop_t*            prop_query;
+    daos_prop_t*            prop_acl = NULL;
+    struct daos_prop_entry* entry;
+    char                    cont_str[DAOS_PROP_LABEL_MAX_LEN];
 
     /*
      * Get all props except the ACL first.
      */
-    prop_query = daos_prop_alloc(1);
-    if (prop_query == NULL)
+    prop_query = daos_prop_alloc(DAOS_PROP_CO_NUM - 1);
+    if (prop_query == NULL) {
         return ENOMEM;
+    }
+
+    prop_query->dpp_entries[0].dpe_type = DAOS_PROP_CO_LABEL;
+    prop_query->dpp_entries[1].dpe_type = DAOS_PROP_CO_LAYOUT_TYPE;
+    prop_query->dpp_entries[2].dpe_type = DAOS_PROP_CO_LAYOUT_VER;
+    prop_query->dpp_entries[3].dpe_type = DAOS_PROP_CO_CSUM;
+    prop_query->dpp_entries[4].dpe_type = DAOS_PROP_CO_CSUM_CHUNK_SIZE;
+    prop_query->dpp_entries[5].dpe_type = DAOS_PROP_CO_CSUM_SERVER_VERIFY;
+    prop_query->dpp_entries[6].dpe_type = DAOS_PROP_CO_REDUN_FAC;
+    prop_query->dpp_entries[7].dpe_type = DAOS_PROP_CO_REDUN_LVL;
+    prop_query->dpp_entries[8].dpe_type = DAOS_PROP_CO_SNAPSHOT_MAX;
+    prop_query->dpp_entries[9].dpe_type = DAOS_PROP_CO_COMPRESS;
+    prop_query->dpp_entries[10].dpe_type = DAOS_PROP_CO_ENCRYPT;
+    prop_query->dpp_entries[11].dpe_type = DAOS_PROP_CO_OWNER;
+    prop_query->dpp_entries[12].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+    prop_query->dpp_entries[13].dpe_type = DAOS_PROP_CO_DEDUP;
+    prop_query->dpp_entries[14].dpe_type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+
+    /* TODO DAOS these as well? */
+    //prop_query->dpp_entries[15].dpe_type = DAOS_PROP_CO_STATUS;
+    //prop_query->dpp_entries[16].dpe_type = DAOS_PROP_CO_ALLOCED_OID;
 
-    prop_query->dpp_entries[0].dpe_type = DAOS_PROP_CO_LAYOUT_TYPE;
     rc = daos_cont_query(cont, NULL, prop_query, NULL);
     if (rc != 0) {
-        MFU_LOG(MFU_LOG_ERR, "failed to query container: %d", rc);
+        MFU_LOG(MFU_LOG_ERR, "failed to query container "DF_RC, DP_RC(rc));
+        rc = 1;
         goto out;
     }
 
-    /* serialize cont property values that are integers */
-    /* DAOS_PROP_CO_LAYOUT_TYPE */
-    len = snprintf(cont_str, 64, "%s", "DAOS_PROP_CO_LAYOUT_TYPE");
-    if (len >= 64) {
-        MFU_LOG(MFU_LOG_ERR, "cont prop string is too long");
+    /* Fetch the ACL separately in case user doesn't have access */
+    rc = daos_cont_get_acl(cont, &prop_acl, NULL);
+    if (rc && rc != -DER_NO_PERM) {
+        MFU_LOG(MFU_LOG_ERR, "failed to query container ACL "DF_RC, DP_RC(rc));
         rc = 1;
         goto out;
     }
-    printf("cont_str: %s\n", cont_str);
-    rc = cont_serialize_prop_uint(hdf5, &prop_query->dpp_entries[0].dpe_val,
-                                  cont_str);
+
+    /* TODO DAOS maybe pass static strings to cont_serialize_prop_str, etc? */
```

Comment:
@dsikich If we just pass a static string like this:
```
cont_serialize_prop_str(hdf5, entry, "DAOS_PROP_CO_LABEL")
```
Then we don't need the local `cont_str` at all and we don't need to use `strncpy`.
But maybe there was a reason why you chose to not do it this way?

Reply:
@daltonbohning no I started doing it that way, then realized I did not need it, and can be removed.

Reply:
@dsikich Great, I'll simplify this then. Thanks!